packages/metro-react-native-babel-preset/src/configs/main.js

```diff
@@ -31,6 +31,7 @@ const defaultPlugins = [
   [require('@babel/plugin-syntax-dynamic-import')],
   [require('@babel/plugin-syntax-export-default-from')],
   ...passthroughSyntaxPlugins,
+  [require('@babel/plugin-transform-named-capturing-groups-regex')],
```

Comment:
I think JavaScriptCore for iOS 11+ has this feature - maybe this plugin should be added in the "extraPlugins" section only `if (isHermes)`?
The Android JavaScriptCore has functionality of about the iOS 12 release, so I expect it also has this feature.

Reply:
Thanks for the review @newyankeecodeshop . I updated the PR with your recommendation.